osu.Game.Rulesets.Mania/ManiaInputManager.cs

```diff
@@ -17,10 +17,13 @@ public ManiaInputManager(RulesetInfo ruleset, int variant)
 
     public enum ManiaAction
     {
-        [Description("Special")]
-        Special,
+        [Description("Special 1")]
+        Special1 = 1,
+        [Description("Special 2")]
+        Special2 = 2,
+
         [Description("Key 1")]
-        Key1 = 10,
+        Key1 = 1000,

```

Comment:
what's this for?

Reply:
Added an XMLDoc for this. Am going to keep the 1000 value after all - I think it's a safer value than 10 at least, when/if we start going into double scratches and whatnot.

Reply:
anything against using `2 << 6` (or `2 << 10` if you don't think 127 options is enough..) instead of `1000`?

Reply:
Bitshifts make me think this is used as flags, where it's not. I can reduce it to 100 if desired. /shrug

Reply:
base 10 makes me think it's there for a reason other than freeing up space. also bit-specific gives the nice property of being able to retrieve keys (minus special keys) by excluding the lower bits.

Reply:
The "nice bit-specific" property you mention isn't going to work in this case, because 2 << 6 + 1 has an LSB of 1. I also don't want to get into limitations that osu-stable mania has as a result of using, essentially `Key1 = 1, Key2 = 2, Key3 = 4, etc...`, so I'm not considering that option.

Reply:
Fair enough. Any reason we can't just add enough special keys now and be done with it?

Reply:
How many do you add? And non-implemented/unused actions? This is just if we want to consider something like that in the future.